src/build.d

```diff
@@ -346,60 +346,41 @@ alias runDmdUnittest = makeDep!((builder, dep) {
-        target = env["G"].buildPath("cxxfrontend").objName;
-
-        command = [ env["CXX"], "-c", sources[0], "-o" ~ target, "-I" ~ env["D"] ] ~ flags["CXXFLAGS"];
-    }
-
-    Dependency cxxUnittestExe; /// Compiles the C++ unittest executable
-    version (Windows) {}
-    else with (cxxUnittestExe)
-    {
-        name = "cxx-unittest";
-        description = "Build the C++ unittests";
-        msg = "(DMD) CXX-UNITTEST";
-
-        deps = [lexer, backend, new DependencyRef(cxxFrontend)];
-        sources = .sources.dmd ~ .sources.root;
-        target = env["G"].buildPath("cxx-unittest").exeName;
+alias runCxxUnittest = makeDep!((runCxxBuilder, runCxxDep) {
+
+    /// Compiles the C++ frontend test files
+    alias cxxFrontend = methodInit!(Dependency, (frontendBuilder, frontendDep) => frontendBuilder
+        .name("cxx-frontend")
+        .description("Build the C++ frontend")
+        .msg("(CXX) CXX-FRONTEND")
+        .sources(srcDir.buildPath("tests", "cxxfrontend.c") ~ .sources.frontendHeaders ~ .sources.dmd ~ .sources.root)
+        .target(env["G"].buildPath("cxxfrontend").objName)
+        .command([ env["CXX"], "-c", frontendDep.sources[0], "-o" ~ frontendDep.target, "-I" ~ env["D"] ] ~ flags["CXXFLAGS"])
+    );
 
-        command = [ env["HOST_DMD_RUN"], "-of=" ~ target, "-vtls", "-J" ~ env["RES"],
+    alias cxxUnittestExe = methodInit!(Dependency, (exeBuilder, exeDep) => exeBuilder
+        .name("cxx-unittest")
+        .description("Build the C++ unittests")
+        .msg("(DMD) CXX-UNITTEST")
+        .deps([lexer, backend, cxxFrontend])
+        .sources(sources.dmd ~ sources.root)
+        .target(env["G"].buildPath("cxx-unittest").exeName)
+        .command([ env["HOST_DMD_RUN"], "-of=" ~ exeDep.target, "-vtls", "-J" ~ env["RES"],
                     "-L-lstdc++", "-version=NoMain"
-        ].chain(
-            flags["DFLAGS"], sources, deps.map!(d => d.target)
-        ).array;
-    }
-
-    Dependency runCxxUnittest; /// Runs the executable created above
-    with (runCxxUnittest)
-    {
-        name = "cxx-unittest";
-        description = "Run the C++ unittests";
-        msg = "(RUN) CXX-UNITTEST";
+            ].chain(
+                flags["DFLAGS"], exeDep.sources, exeDep.deps.map!(d => d.target)
+            ).array)
+    );
 
-        version (Windows)
-        {
-            commandFunction = { assert(0, "Running the C++ unittests is not supported on Windows yet"); };
-        }
-        else
-        {
-            auto cxxUnittest = new DependencyRef(cxxUnittestExe);
-            deps = [cxxUnittest];
-            command = [cxxUnittest.target];
-        }
-    }
-    return new DependencyRef(runCxxUnittest);
+    runCxxBuilder
+        .name("cxx-unittest")
+        .description("Run the C++ unittests")
+        .msg("(RUN) CXX-UNITTEST");
+    version (Windows) runCxxBuilder
+        .commandFunction({ assert(0, "Running the C++ unittests is not supported on Windows yet"); });
```

Comment:
Use `enforce`, instead of `assert` as it's more likely that we're validating the input of the `build.d` user, instead catching a programmer bug.